client/v3/watch.go

```diff
@@ -581,10 +581,12 @@ func (w *watchGrpcStream) run() {
 			switch {
 			case pbresp.Created:
 				// response to head of queue creation
-				if ws := w.resuming[0]; ws != nil {
-					w.addSubstream(pbresp, ws)
-					w.dispatchEvent(pbresp)
-					w.resuming[0] = nil
+				if len(w.resuming) != 0 {
```

Comment:
Sorry for being late. Did you see issues related to this conditional branch? I think it is an independent issue of clientv3 (not an issue of gRPC proxy). Could you open a new PR just for submitting the change?

Reply:
I hope keep this code, otherwise this will make a panic when etcd server all down and recover, this PR will not complete.

Reply:
yeah, I meant it's great if you can create an individual PR for this change at first and merge to the master branch. I think the change is important not only for grpc proxy but also for the entire etcd components.

Reply:
https://github.com/etcd-io/etcd/pull/12925/files

Reply:
thanks!